test/test_document_store.py

```diff
@@ -1,16 +1,15 @@
-from unittest import mock
-import uuid
-import math
 import numpy as np
 import pandas as pd
 import pytest
+import json
+import responses
```

Comment:
Just a curiosity here: what's the advantage of `responses` over a hand-made mock of `requests`? I'm surprised you need those stream files to make this all work. With a hand-made mock, you could probably get away with a few dictionaries I think.

If you decide to implement `_query_dc` method from the comments above, you might even just mock that method and forget about the `requests` mock at all... :grin:

Reply:
responses allows for easy request handling with various finegrained matchers like header_matcher, json_param_matcher without having to inspect the request manually. In total, it should have saved me at least half the lines of code I would have needed if I implemented the same with bare requests mock or a previous method. 
But I can see, that it looks pretty confusing inside the fixture. Maybe I can do better there.